SparkPost/2web2ui
# FE-786 - Add Cypress integration tests for Recipient Validation (RV)

[FE-786](https://jira.int.messagesystems.com/browse/FE-786)

### What Changed
- Adds `cy.stubRequest` command to help reduce the boilerplate required to intercept a network request - @bkemper (...you mentioned adding this last time and I disagreed then and since changed my mind 😅)
- Adds Cypress integration tests for RV features
- Adds a whole bunch of RV-related fixtures
- Adds `cypress-file-upload` package to help mock file uploading in a Cypress test
- Adds screen reader only form field label to the single recipient validation form both as an a11y and testing improvement (`/recipient-validation/single`)
- Some Prettier re-formatting that came along for the ride

### How To Test
1. Run through the RV feature and verify that the written tests cover common use-cases:
  - Try uploading a list of recipients to validate
  - Try validating a single email address
2. Verify that all tests are passing (Travis should do this for you)
3. Verify that test descriptions and assertions make sense and document expectations/requirements

### To Do
- [x] Add tests for the list validation confirmation page
- [x] Figure out a way to test polling (maybe using `cy.wait()`?) - **Decided against this** - the Jest tests are already testing this behavior, which tend to do a better job checking for network requests.
